src/controllers/user.controller.js

```diff
@@ -78,4 +81,89 @@ const loginUser = async (req, res, next) => {
     }
   })(req, res, next);
 };
-export { registerUser, loginUser };
+// export { registerUser, loginUser };
+
+
+const apiKey = process.env.API_KEY;
+sgMail.setApiKey(apiKey);
+
+ const resetEmail = async (req, res) => {
+  try {
+  
+   const userEml = req.body.email;
+   const userEmail = await findUser(userEml);
+    if (userEmail == false) {
+      res.status(404).json("User not found");
+    } else {
+      const userToken = generateToken(userEmail.email, { expiresIn: '10m' });
+
+      const message = {
+        to: req.body.email,
+        from: {
+          name: "DESTRUCTORS",
+          email: process.env.SEND_EMAIL
+        },
+        subject: 'Reset password',
+
+        html: `<p stlye="font-size: 30px"><strong> Hi <br> <br>
+            Please click on this link below to reset your password:<br> ${userToken}><br>Remember, beware of scams and keep this one-time verification link confidential.<br>
+            Thanks, </strong><br> DESTRUCTORS </p>`
+
+      };
+      sgMail
+        .send(message)
+        .then((res) => console.log('email sent...'))
+        .catch((error) => {
+          console.log(error.message)
+        });
```

Comment:
@katros1 , can we have a main function that sends an email and can be used everywhere? it can take **subject** and  **body** and **data** which is parameters. In your case, data can be **userToken** etc.

ie everywhere we want to send an email we can call your function.

Reply:
@Abdulkeza thank you for this, Now I finish to work on it, you can check if  I  done it perfectly